### PR TITLE
Update project to Kotlin 1.9

### PR DESCRIPTION
**Pull request by zsmb13**
Updates Kotlin to the latest version, adjusting KSP and Compose compiler versions accordingly. Updates Dagger to [2.47](https://github.com/google/dagger/releases/tag/dagger-2.47) get it to a newer version of kotlinx-metadata.

### CODE REVIEW

File: app/src/androidTest/java/com/google/samples/apps/nowinandroid/ui/NiaAppStateTest.kt [hunk 1]
```diff
@@ -183,7 +183,7 @@ class NiaAppStateTest {
 @Composable
 private fun rememberTestNavController(): TestNavHostController {
     val context = LocalContext.current
-    val navController = remember {
+    return remember<TestNavHostController> {
```
Comment (zsmb13) [inline app/src/androidTest/java/com/google/samples/apps/nowinandroid/ui/NiaAppStateTest.kt:186]: Required due to a lint bug https://issuetracker.google.com/291845885

**Comment by SimonMarquis** [comment]
Hi @zsmb13 , there is currently another PR tracking the Kotlin 1.9 update #752 and there is still a blocking issue.

**Comment by zsmb13** [comment]
Ah, sorry @SimonMarquis, let me bump Compose to stable over there and use that PR/branch then.

**Comment by SimonMarquis** [comment]
No worries, I can take care of that if you want me to. Although, Hilt could have been merged without updating the whole Kotlin stack.

**Pull request closed without merging**